libr/anal/var.c

```diff
@@ -696,11 +723,8 @@ R_API void r_anal_extract_rarg(RAnal *anal, RAnalOp *op, RAnalFunction *fcn, int
 	}
 	for (i = 0; i < max_count; i++) {
 		const char *regname = r_anal_cc_arg (anal, fcn->cc, i);
-		// reg_set enusres we only extract first-read argument reg
-		if (!reg_set [i] && regname) {
```

Comment:
I think you should move this code in the else block of the reg below and check for opsreg and opdreg to see if they are null to use the old method. This cond is the one that may be taken back

Reply:
Yep, that will fix behaviour for failing test. I’ll try to change it today.